sdk/communication/azure-communication-identity/azure/communication/identity/_communication_identity_client.py

```diff
@@ -88,7 +86,7 @@ def from_connection_string(
                 :caption: Creating the CommunicationIdentityClient from a connection string.
         """
         endpoint, access_key = parse_connection_str(conn_str)
-        return cls(endpoint, access_key, **kwargs)
+        return cls(endpoint, access_key, **kwargs)  # type: ignore
```

Comment:
It gives mypy error : `Argument 2 to "CommunicationIdentityClient" has incompatible type "str"; expected "TokenCredential | AzureKeyCredential"  [arg-type]`
In the __init__() method, it is defined as `credential,  # type: Union[TokenCredential, AzureKeyCredential]`
If I add str to this Union, it will be treated as a breaking change. 

Could you please suggest what can be done here ?

Reply:
@kristapratico Could you please help me with the solution for this mypy error ?

Reply:
Yes, there is logic downstream in method `get_authentication_policy` to handle string credential. 
Wrapping the string in AzureKeyCredential didn't worked.
Added type: ignore with a comment.